datasets/hate_speech_offensive/hate_speech_offensive.py

```diff
@@ -0,0 +1,100 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""An annotated dataset for hate speech and offensive language detection on tweets."""
+
+from __future__ import absolute_import, division, print_function
+
+import csv
+
+import datasets
+
+
+_CITATION = """\
+@inproceedings{hateoffensive,
+title = {Automated Hate Speech Detection and the Problem of Offensive Language},
+author = {Davidson, Thomas and Warmsley, Dana and Macy, Michael and Weber, Ingmar},
+booktitle = {Proceedings of the 11th International AAAI Conference on Web and Social Media},
+series = {ICWSM '17},
+year = {2017},
+location = {Montreal, Canada},
+pages = {512-515}
+}
+"""
+
+_DESCRIPTION = """\
+An annotated dataset for hate speech and offensive language detection on tweets.
+"""
+
+_HOMEPAGE = "https://github.com/t-davidson/hate-speech-and-offensive-language"
+
+_LICENSE = "Unknown"
```

Comment:
The github repo is licensed under MIT license: https://github.com/t-davidson/hate-speech-and-offensive-language/blob/master/LICENSE